substrate-node/pallets/pallet-smart-contract/src/lib.rs

```diff
@@ -1413,6 +1405,11 @@ impl<T: Config> Pallet<T> {
             // balance we can only transfer out the remaining balance
             // https://github.com/threefoldtech/tfchain/issues/479
             let twin_balance = Self::get_usable_balance(&twin.account_id);
+            log::info!(
+                "twin balance {:?} contract lock amount {:?}",
+                twin_balance,
+                contract_lock.amount_locked
+            );
```

Comment:
Ok, the idea here is, ideally, to make the user pay for `amount_locked` and if he has not enough funds to drain its account, correct?
And then reset its `amount_locked` to `0`

Let s imagine an extreme case where `usuable_balance = 1` and `amount_locked = 1000`
So the user will effectively pay `1` and the remaining `999` due will vanish
Is that what we want?

Reply:
yes, but this is only a viable flow in the case that a contract is in grace period or a contract got canceled. We cannot take more funds out of the users wallet then whatever he has left.

In the normal billing case, the user will always pay whatever he's due.